src/open_inwoner/cms/cases/views/cases.py

```diff
@@ -6,116 +6,81 @@
 
-        # anchors are needed here as well for rendering the mobile ones
-        context["anchors"] = self.get_anchors()
-        context["hxget"] = reverse("cases:open_cases_content")
+        context["hxget"] = reverse("cases:cases_content")
         return context
 
 
-class InnerOpenCaseListView(
-    RequiresHtmxMixin, CommonPageMixin, CaseAccessMixin, CaseListMixin, TemplateView
+class InnerCaseListView(
+    RequiresHtmxMixin,
+    CommonPageMixin,
+    CaseAccessMixin,
+    CaseLogMixin,
+    PaginationMixin,
+    TemplateView,
 ):
+    """View on case list"""
+
     template_name = "pages/cases/list_inner.html"
+    paginate_by = 9
 
     def page_title(self):
-        return _("Lopende aanvragen")
+        return _("Mijn aanvragen")
 
     def get_cases(self):
-        all_cases = super().get_cases()
+        raw_cases = fetch_cases(self.request.user.bsn)
+        preprocessed_cases = preprocess_data(raw_cases)
+        preprocessed_cases.sort(key=lambda case: case.startdatum, reverse=True)
+        return preprocessed_cases
 
-        cases = [case for case in all_cases if not case.einddatum]
-        cases.sort(key=lambda case: case.startdatum, reverse=True)
-        return cases
-
-    def get_anchors(self) -> list:
-        return [
-            (reverse("cases:open_submissions"), _("Openstaande aanvragen")),
-            ("#cases", _("Lopende aanvragen")),
-            (reverse("cases:closed_cases"), _("Afgeronde aanvragen")),
-        ]
+    def get_submissions(self):
+        subs = fetch_open_submissions(self.request.user.bsn)
+        subs.sort(key=lambda sub: sub.datum_laatste_wijziging, reverse=True)
+        return subs
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        context["hxget"] = reverse("cases:open_cases_content")
-        return context
+        config = OpenZaakConfig.get_solo()
 
+        # update ctx with cases + submissions
+        preprocessed_cases = self.get_cases()
+        open_submissions = self.get_submissions()
+        paginator_dict = self.paginate_with_context(
+            [*open_submissions, *preprocessed_cases]
+        )
+        case_dicts = [case.process_data() for case in paginator_dict["object_list"]]
 
-class OuterClosedCaseListView(
-    OuterCaseAccessMixin, CommonPageMixin, BaseBreadcrumbMixin, TemplateView
-):
-    template_name = "pages/cases/list_outer.html"
-
-    @cached_property
-    def crumbs(self):
-        return [(_("Mijn aanvragen"), reverse("cases:closed_cases"))]
-
-    def page_title(self):
-        return _("Afgeronde aanvragen")
+        context["cases"] = case_dicts
+        context.update(paginator_dict)
 
-    def get_anchors(self) -> list:
-        return [
-            (reverse("cases:open_submissions"), _("Openstaande aanvragen")),
-            (reverse("cases:open_cases"), _("Lopende aanvragen")),
-            ("#cases", _("Afgeronde aanvragen")),
-        ]
+        self.log_access_cases(case_dicts)
```

Comment:
I note this now logs both cases and submissions.

Reply:
Yes, submissions are now cases too. I created a `Protocol` to document this but only added the type hints above (lines 61-62) after Steven's suggestion.